### PR TITLE
[python-package] Implement __eq__ for CatBoost* classes

### PR DESCRIPTION
**Pull request by daskol**
Another one attempt to contribute something (see #645 and #650).

### CODE REVIEW

**Comment by annaveronika** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@annaveronika, internal review request created: 701573

File: catboost/python-package/ut/medium/test.py [hunk 1]
```diff
@@ -3664,3 +3664,34 @@ def test_eval_set_with_no_target_with_eval_metric(task_type):
     )
     with pytest.raises(CatboostError):
         model.fit(train_pool, eval_set=eval_set_pool)
+
+
+def test_model_comparison():
+    def fit_model(value):
```
Comment (andrey-khropov) [inline catboost/python-package/ut/medium/test.py]: After changing to adult pool this function no longer depends on value so both model1 and model2 will be the same. I propose make 'iterations' a parameter and check 3 vs 5 iterations, for example.

Comment (daskol) [reply catboost/python-package/ut/medium/test.py]: Done.

**Pull request merged**